caveclient/chunkedgraph.py

```diff
@@ -198,6 +198,18 @@ def default_url_mapping(self):
     def table_name(self):
         return self._table_name
 
+    @property
+    def timestamp(self) -> Optional[datetime.datetime]:
+        """Default timestamp to use in queries that accept a `timestamp` argument.
+        If None, equivalent to querying the current state of the chunkedgraph."""
+        return self._default_timestamp
+
+    @timestamp.setter
+    def timestamp(self, timestamp: Optional[datetime.datetime]):
+        if (timestamp is not None) and (not isinstance(timestamp, datetime.datetime)):
+            raise ValueError("Timestamp must be a datetime.datetime object")
```

Comment:
Unsure how much validation makes sense here. The current behavior is [no validation](https://github.com/CAVEconnectome/CAVEclient/blob/3208beae570a7eac11a7501329ce20ca0259ba8b/caveclient/chunkedgraph.py#L189), but in materialization the version is at least [checked against available versions](https://github.com/CAVEconnectome/CAVEclient/blob/3208beae570a7eac11a7501329ce20ca0259ba8b/caveclient/materializationengine.py#L293). I could imagine doing something similar here to see if the timestamp is even after the beginning of the database or something like that?